src/main/kotlin/io/embrace/android/intellij/plugin/ui/forms/EmbraceIntegrationForm.kt

```diff
@@ -81,9 +81,8 @@ internal class EmbraceIntegrationForm(
         panel.add(EmbLabel("step1Description".text(), TextStyle.BODY))
         panel.add(Box.createVerticalStrut(VERTICAL_SPACE))
         panel.add(EmbButton("btnConnect".text()) {
-            dataProvider.startServer()
+            dataProvider.startServer(etAppId, etToken)
             dataProvider.openDashboard()
-            dataProvider.openBrowserAtCallback()
```

Comment:
I would move these to a single function inside the dataProvider. This way, the data layer will be responsible for executing these functions in the desired order, and the UI remains simple.